File: algorithms/7.py
# https://leetcode.com/problems/reverse-integer/


class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        r = 0
        n = abs(x)
        while n > 0:
            r = 10 * r + n % 10
            n //= 10

        if r > pow(2, 31) - 1:
            return 0
        else:
            return r if x >= 0 else -r

File: algorithms/7.swift
// https://leetcode.com/problems/reverse-integer/

class Solution {
    func reverse(x: Int) -> Int {
        var r = 0
        var n = abs(x)
        while n > 0 {
            r = 10*r + n%10
            if Double(r) > pow(2.0, 31.0)-1 { return 0 }
            n /= 10
        }

        return (x>=0) ? r : -r
    }
}
